### PR TITLE
Fix lm_eval.tasks' has no attribute 'initialize_tasks' error

### LINKED ISSUE #422: evaluation
### System Info

 python eval.py --model hf --model_args pretrained=/mnt/sdb/zjk/llama2/llama-recipes/Llama-2-7b-hf,dtype="float",peft=/mnt/sdb/zjk/llama2/llama2-lora --tasks hellaswag --num_fewshot 10  --device cuda:0 --batch_size 8 
error：
2024-04-02:06:58:12,156 ERROR    [eval.py:226] An error occurred during evaluation: module 'lm_eval.tasks' has no attribute 'initialize_tasks'

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### 🐛 Describe the bug

evaluation was changed？
how can i evaluate finetuned model with lora now？

### Error logs

2024-04-02:06:58:12,156 ERROR    [eval.py:226] An error occurred during evaluation: module 'lm_eval.tasks' has no attribute 'initialize_tasks'

### Expected behavior

evaluate finetuned model with lora and the accuracy of specific dataset with cola

### PR DESCRIPTION
**Pull request by KeitaW**
# What does this PR do?

Fixes #422 

lm-evaluation-harness has introduced breaking changes causing `lm_eval.tasks' has no attribute 'initialize_tasks'` error in the current `recipes/evaluation/eval.py`. This PR is aiming to resolve the issue by adopting new TaskManager API.
cf. https://github.com/EleutherAI/lm-evaluation-harness/releases

## Feature/Issue validation/testing

Please describe the tests that you ran to verify your changes and relevant result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

```bash
python/eval.py \
     --model hf --model_args pretrained=meta-llama/Meta-Llama-3-8B --tasks hellaswag --device cuda:0   --batch_size 8"
```

yields

```bash
Map: 100%|██████████| 10042/10042 [00:01<00:00, 8628.17 examples/s]
0: 2024-05-06:12:44:13,769 INFO     [task.py:395] Building contexts for hellaswag on rank 0...
 62%|██████▏   | 6276/10042 [00:01<00:01, 2603.76it/s]
100%|██████████| 10042/10042 [00:02<00:00, 3578.18it/s]
0: 2024-05-06:12:44:17,334 INFO     [evaluator.py:393] Running loglikelihood requests
...
0: 2024-05-06:12:47:36,076 INFO     [eval.py:39] {
0:   "results": {
0:     "hellaswag": {
0:       "acc,none": 0.6015733917546305,
0:       "acc_stderr,none": 0.004885735963346892,
0:       "acc_norm,none": 0.7905795658235412,
0:       "acc_norm_stderr,none": 0.004060633907027273,
0:       "alias": "hellaswag"
0:     }
0:   },
...
```

## Before submitting
- [] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/facebookresearch/llama-recipes/blob/main/CONTRIBUTING.md#pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes?  
- [ ] Did you write any new necessary tests?

Thanks for contributing 🎉!

### CODE REVIEW

**Review by HamidShojanazeri — APPROVED** [review]
Thanks @KeitaW for the PR.
